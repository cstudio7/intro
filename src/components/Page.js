import React, { Component } from 'react';
import {
    StyleSheet,
    Dimensions,
    View,
    StatusBar
} from 'react-native';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import tinycolor from 'tinycolor2';
import PageContent from './PageContent';
import Paginator from './Paginator';
import Skip from './Skip';

export class Page extends Component {
    state = {
        offsetX: 0
    }


    offsetAlongX = (direction) => {
        // calculate offset along X axis
        // We get this direction from the child = Paginator.js
        // We will pass the value of offset to PageContent.js
        // from where we scrollTo offset
        const { width } = Dimensions.get('window');
        let offsetX = 0;
        if (direction === 'next') {
            offsetX = this.state.offsetX + width;
        } else {
            offsetX = this.state.offsetX - width;
        }
        this.setState({ offsetX });
    }

    lightBackground() {
        // detect if the background color is light or dark
        const color = tinycolor(this.props.pages[this.props.currentIndex].backgroundColor)
        return color.getBrightness() > 180;
    }

    render() {
        return (
            <View
                style={[styles.container, {backgroundColor: this.props.pages[this.props.currentIndex].backgroundColor}]}
            >
                <StatusBar
                    backgroundColor={this.props.pages[this.props.currentIndex].backgroundColor}
                    barStyle={this.lightBackground() ? 'dark-content' : 'light-content'}
                />
                <Skip
                    numberOfDots={this.props.numberOfDots}
                    lightBackground={this.lightBackground()}
                    onSkip={() => console.log('Alert: Navigate to the root App')}
                />
                <PageContent
                    pages={this.props.pages}
                    offsetX={this.state.offsetX} //offset gotten
                    lightBackground={this.lightBackground()}
                />
                <Paginator
                    numberOfDots={this.props.numberOfDots}
                    pages={this.props.pages}
                    scroll={this.offsetAlongX}
                    lightBackground={this.lightBackground()}
                />
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'space-between',
    },
});

const mapStateToProps = (state) => {
    return {
        currentIndex: state.page.currentIndex
    }
}

Page.propTypes = {
    pages: PropTypes.arrayOf(PropTypes.shape({
        backgroundColor: PropTypes.string.isRequired,
        image: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        subtitle: PropTypes.string.isRequired
    })).isRequired,
    numberOfDots: PropTypes.number.isRequired
};

//Page.defaultProps = {
//
//}

export default connect(mapStateToProps)(Page);
