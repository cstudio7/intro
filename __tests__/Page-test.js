import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import Enzyme, {shallow} from 'enzyme';
import {Provider} from 'react-redux';
import Adapter from 'enzyme-adapter-react-16';

import Page from '../src/components/Page.js';



Enzyme.configure({ adapter: new Adapter() });


const mockStore = configureMockStore()


describe('Page wrapped with redux is ok', () => {
    let wrapper, store;

    const initialState = {currentIndex: 0}

    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = shallow(<Page store={store} />);
    });

    it('renders alright', () => {
        expect(wrapper).toMatchSnapshot()
    });

    it('gets redux props', () => {
        expect(wrapper.props().currentIndex).toEqual(0)
    });

});
