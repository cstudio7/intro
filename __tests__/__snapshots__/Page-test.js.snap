// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders as expected 1`] = `null`;

exports[`Page wrapped with redux connect works handles redux well 1`] = `undefined`;

exports[`Page wrapped with redux is ok renders alright 1`] = `undefined`;
